qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockIT.java

```diff
@@ -0,0 +1,236 @@
+
+    // when - producing records
+    final var deployResult =
+        zeebeClient.newDeployCommand().addProcessModel(process, "process.bpmn").send().join();
+    zeebeClient
+        .newCreateInstanceCommand()
+        .processDefinitionKey(
+            deployResult.getProcesses().stream()
+                .findFirst()
+                .orElseThrow()
+                .getProcessDefinitionKey())
+        .send()
+        .join();
+
+    // then - records are exported with a timestamp matching the offset time
+    Awaitility.await()
+        .untilAsserted(
+            () -> {
+              final var records = searchExportedRecords();
+              assertThat(records).isNotNull();
+              assertThat(records.size())
+                  .isEqualTo(10); // deploy + instance produces exactly 10 records
+              assertThat(records)
+                  .allSatisfy(
+                      (record) -> {
+                        final var timestamp = Instant.ofEpochMilli(record.getTimestamp());
+                        final var afterRecords = Instant.now();
+                        assertThat(timestamp).isBefore(afterRecords.plus(offsetZeebeTime));
+                        assertThat(timestamp).isAfter(beforeRecords.plus(offsetZeebeTime));
+                      });
+            });
+  }
+
+  private List<AbstractRecord<?>> searchExportedRecords() throws IOException, InterruptedException {
+    final var uri =
+        URI.create(
+            String.format(
+                "http://%s/%s*/_search",
+                elasticsearchContainer.getHttpHostAddress(), INDEX_PREFIX));
+    final var request = HttpRequest.newBuilder(uri).method("POST", BodyPublishers.noBody()).build();
+    final var response = httpClient.send(request, BodyHandlers.ofInputStream());
+    final var result = mapper.readValue(response.body(), EsSearchResponseDto.class);
+    if (result.documentsWrapper == null) {
+      return null;
+    }
+    return result.documentsWrapper.documents.stream()
+        .map(esDocumentDto -> esDocumentDto.record)
+        .collect(Collectors.toList());
+  }
+
+  private Instant pinZeebeTime() throws IOException, InterruptedException {
+    final var pinAt = Instant.now().minus(Duration.ofDays(1));
+    final var body =
+        BodyPublishers.ofString(String.format("{\"epochMilli\": %s}", pinAt.toEpochMilli()));
+    zeebeRequest("/actuator/clock/pin", body);
+    return pinAt;
+  }
+
+  private Duration offsetZeebeTime() throws IOException, InterruptedException {
+    final var offsetBy = Duration.ofHours(3);
+    final var body =
+        BodyPublishers.ofString(String.format("{\"offsetMilli\": %s}", offsetBy.toMillis()));
+    zeebeRequest("/actuator/clock/add", body);
+    return offsetBy;
+  }
+
+  private void zeebeRequest(final String endpoint, final BodyPublisher body)
+      throws IOException, InterruptedException {
+    final var fullEndpoint =
```

Comment:
:wrench: There's a `PartitionsActuatorClient` in the `test-util` module. We could think about adding another one for the clock there. Note that the design of the existing client is questionable (I should know, I wrote it :smile:), so you don't need to copy it exactly, feel free to improve it :slightly_smiling_face:

Reply:
I wrote a bare-minimum `ClockActuatorClient` that turns out to be fairly similar to your `PartitionsActuatorClient` :laughing: